feature/container/containerz/tests/container_lifecycle/containerz_test.go

```diff
@@ -107,182 +127,689 @@ func stopContainer(ctx context.Context, t *testing.T, cli *client.Client) {
+		if err != nil {
+			// Case 1: Logs() itself returns an error
+			t.Logf("Got expected error when retrieving logs for non-existent instance %s: %v", nonExistentInstanceName, err)
+			s, _ := status.FromError(err)
+			// Docker daemon might return Unknown for "No such container" when accessed via an API like containerz.
+			if s.Code() != codes.NotFound && s.Code() != codes.Unknown {
+				t.Errorf("Expected codes.NotFound or codes.Unknown for non-existent instance %s, but got %s.", nonExistentInstanceName, s.Code())
+			}
+			if logCh != nil {
+				t.Errorf("Expected nil logCh when cli.Logs returns an error for non-existent instance %s, but got %v", nonExistentInstanceName, logCh)
+			}
+			return
+		}
+
+		// Case 2: Logs() returns (channel, nil). Error should come via channel.
+		if logCh == nil {
+			t.Fatalf("cli.Logs for non-existent instance %s returned nil channel and nil error, expected error via channel or direct error.", nonExistentInstanceName)
+		}
+
+		t.Logf("cli.Logs for non-existent instance %s returned nil error, expecting error via channel.", nonExistentInstanceName)
+
+		// Define a timeout for receiving from the channel.
+		// This prevents the test from hanging indefinitely if the channel never sends.
+		const channelReadTimeout = 10 * time.Second
+		timer := time.NewTimer(channelReadTimeout)
+		defer timer.Stop()
+
+		select {
+		case msg, ok := <-logCh:
+			if !ok {
+				// Channel was closed without sending any message.
+				t.Errorf("Expected an error message on the log channel for non-existent instance %s, but channel closed without sending a message.", nonExistentInstanceName)
+			} else {
+				// A message was received.
+				if msg.Error != nil {
+					t.Logf("Got expected error from log channel for non-existent instance %s: %v", nonExistentInstanceName, msg.Error)
+					s, _ := status.FromError(msg.Error)
+					if s.Code() != codes.NotFound && s.Code() != codes.Unknown {
+						t.Errorf("Expected codes.NotFound or codes.Unknown from channel for non-existent instance %s, but got %s.", nonExistentInstanceName, s.Code())
+					}
+					// Test considers this path as having found the expected error.
+				} else {
+					// An actual log message was received, which is an error for this test case.
+					t.Errorf("Received unexpected log message '%s' for non-existent instance %s when expecting an error.", msg.Msg, nonExistentInstanceName)
+				}
+			}
+		case <-timer.C:
+			// Timeout occurred.
+			t.Errorf("Timed out waiting for a message (expected error) on the log channel for non-existent instance %s after %v.", nonExistentInstanceName, channelReadTimeout)
+		}
+	})
+
+	// Negative Test: Attempt to retrieve logs from a stopped container instance.
+	t.Run("LogsFromStoppedInstance", func(t *testing.T) {
+		stoppedInstanceName := "test-instance-for-stopped-logs"
+		localImageName := imageName
+		localImageTag := "latest"
+
+		// 1. Ensure clean state for this specific container instance.
```

Comment:
For these container setup functions -- can we pull them out into a common function somewhere? It seems like we could re-use a call here across different cases to make the tests a bit more succinct.